### ISSUE TITLE
Modifiers only process properties with separators

### ISSUE THREAD

**Issue by ackushiw** [rhalff/dot-object#38]
Modifiers on root properties do not process the transformation

```javascript
var dot = require('dot-object');

var row = {
  'nr': 200,
  'nested.nr': 200,
};

var mods = {
  'nr': [val => val * 2],
  'nested.nr': [val => val * 2]
};

dot.object(row, mods);

console.log(row);
```
Result:
```
{
  nr: 200,
  nested: {
    nr: 400
  }
}
```
Expected:
```
{
  nr: 400,
  nested: {
     nr: 400
  }
}
```

**Comment by rhalff** [comment]
This should work if `override` is set to true.

There is also a test preventing this:
```
should not process non dot value with modifier when override is false
```

This test however doesn't make much sense to me now and causes unexpected and even erroneous behavior, so I'll update the test and remove the requirement to set `override` to true.

**Comment by rhalff** [comment]
Fixed in new version released as v1.9.0
